src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,50 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+
+  width: (7 * $day-size) + (6 * $day-gap)
+    + (2 * $calendar-padding);
+
+  padding: 25px $calendar-padding;
+
+  @each $day, $number in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: ($number * $day-size) + ($day-gap * $number);
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} {
+      .calendar__date:nth-child(n + #{$index + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__date {
+    width: #{$day-size};
+    height: #{$day-size};
+    display: flex;
+    justify-content: center;
+
+    transition: background-color 0.5s, transform 0.5s;
+    border: 1px solid black;
+    background-color: #eee;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: "#{$day}";
+        align-self: center;
+        font-size: 30px;
```

Comment:
"not critical, but it would be cleaner to move all these to &__day
and just leave the variable part inside @for" this was Roman's review for my code